### PR TITLE
Improve some more docs in std.array

### PR DESCRIPTION
**Pull request by JackStouffer**

### CODE REVIEW

File: std/array.d [hunk 1]
```diff
@@ -2240,9 +2243,15 @@ if (isInputRange!Range &&
...
-    (inclusive) to $(D to) (exclusive) with the range $(D stuff). Expands or
+    Replaces elements from `array` with indices ranging from `from`
+    (inclusive) to `to` (exclusive) with the range `stuff`. Expands or
     shrinks the array as needed.
+
+    Params:
+        array = the _array to scan
+        from = the starting index
+        to = the ending index
+        stuff = the items to replace in-between `from` and `to`
```
Comment (wilzbach) [inline std/array.d]: How about renaming the parameter `stuff` to something meaningful like `items`, `range` or `replacements`?

File: std/array.d [hunk 2]
```diff
@@ -2652,8 +2661,18 @@ if (isDynamicArray!(E[]) &&
 }
 
 /++
-    Returns a new array that is $(D s) with $(D slice) replaced by
-    $(D replacement[]).
+    Creates a new array such that the items in `slice` are replace with the
```
Comment (wilzbach) [inline std/array.d]: s/replace/replaced

File: std/array.d [hunk 3]
```diff
@@ -2652,8 +2661,18 @@ if (isDynamicArray!(E[]) &&
...
+    items in `replacement`. `slice` and replacement do not need to be the
```
Comment (wilzbach) [inline std/array.d]: For consistency I would put the second occurrence of replacement in backticks as well.
Moreover I think a `length` is a property, so a verb denominating ownership like "have" or "share" would be more appropriate.

File: std/array.d [hunk 4]
```diff
@@ -2652,8 +2661,18 @@ if (isDynamicArray!(E[]) &&
...
+    Params:
+        s = the base of the new array
+        slice = the slice of `s` to be replaced
+        replacement = the items to replace `slice` with
+
+    Returns:
+        A new array that is `s` with `slice` replaced by
+        `replacement[]`.
  +/
 inout(T)[] replaceSlice(T)(inout(T)[] s, in T[] slice, in T[] replacement)
```
Comment (wilzbach) [inline std/array.d]: It seems to me that we are in desperate need of a guideline for variable naming. While `r` is common for ranges, `s` for arrays is not :P

File: std/array.d [hunk 5]
```diff
@@ -2767,17 +2786,17 @@ if (isDynamicArray!A)
     }
 
     /**
-     * Returns the capacity of the array (the maximum number of elements the
-     * managed array can accommodate before triggering a reallocation).  If any
-     * appending will reallocate, $(D capacity) returns $(D 0).
+     * Returns: the capacity of the array (the maximum number of elements the
+     * managed array can accommodate before triggering a reallocation). If any
+     * appending will reallocate, returns `0`.
```
Comment (wilzbach) [inline std/array.d]: How about: `0` will be returned?

File: std/array.d [hunk 6]
```diff
@@ -2767,17 +2786,17 @@ if (isDynamicArray!A)
...
      */
     @property size_t capacity() const @safe pure nothrow
     {
         return _data ? _data.capacity : 0;
     }
 
     /**
-     * Returns the managed array.
+     * Returns: The managed array.
```
Comment (wilzbach) [inline std/array.d]: FWIW you use "the" after `Returns:` in the change above. It might make sense to agree on a common convention here as well.

Comment (quickfur) [inline std/array.d]: I think it's fine.

File: std/array.d [hunk 7]
```diff
@@ -2652,8 +2661,18 @@ if (isDynamicArray!(E[]) &&
...
 
 /++
-    Returns a new array that is $(D s) with $(D slice) replaced by
-    $(D replacement[]).
+    Creates a new array such that the items in `slice` are replaced with the
+    items in `replacement`. `slice` and `replacement` do not need to be the
+    same length. The result will grow or shrink based on the items given.
+
+    Params:
+        s = the base of the new array
```
Comment (ntrel) [inline std/array.d]: I think 'base' is confusing, maybe use `= the original array` or 'source array'.

**Pull request merged**